src/pykeen/nn/node_piece.py

```diff
@@ -685,6 +673,130 @@ def __call__(
+    base=PrecomputedTokenizerLoader,
+    default=GalkinPickleLoader,
+)
+
+
+class PrecomputedPoolTokenizer(Tokenizer):
+    """A tokenizer using externally precomputed tokenization."""
+
+    @classmethod
+    def _load_pool(
+        cls,
+        *,
+        path: Optional[pathlib.Path] = None,
+        url: Optional[str] = None,
+        download_kwargs: OptionalKwargs = None,
+        pool: Optional[Mapping[int, Collection[int]]] = None,
+        loader: HintOrType[PrecomputedTokenizerLoader] = None,
+    ) -> Tuple[Mapping[int, Collection[int]], int]:
+        """Load a precomputed pool via one of the supported ways."""
+        if pool is not None:
+            return pool, max(c for candidates in pool.values() for c in candidates) + 1 + 1  # +1 for padding
+        if url is not None and path is None:
+            module = PYKEEN_MODULE.submodule(__name__, tokenizer_resolver.normalize_cls(cls=cls))
+            path = module.ensure(url=url, download_kwargs=download_kwargs)
+        if path is None:
+            raise ValueError("Must provide at least one of pool, path, or url.")
+
+        if not path.is_file():
+            raise FileNotFoundError(path)
+        logger.info(f"Loading precomputed pools from {path}")
+        return precomputed_tokenizer_loader_resolver.make(loader)(path=path)
+
+    def __init__(
+        self,
+        *,
+        path: Optional[pathlib.Path] = None,
+        url: Optional[str] = None,
+        download_kwargs: OptionalKwargs = None,
+        pool: Optional[Mapping[int, Collection[int]]] = None,
+    ):
+        """
+        Initialize the tokenizer.
+
+        .. note ::
+            the preference order for loading the precomputed pools is (1) from the given pool (2) from the given path,
+            and (3) by downloading from the given url
+
+        :param path:
+            a path for a file containing the precomputed pools
+        :param url:
+            an url to download the file with precomputed pools from
+        :param download_kwargs:
+            additional download parameters, passed to pystow.Module.ensure
+        :param pool:
+            the precomputed pools.
+
+        """
+        self.pool, self.vocabulary_size = self._load_pool(
+            path=path, url=url, pool=pool, download_kwargs=download_kwargs
+        )
+        # verify pool
+        if set(self.pool.keys()) != set(range(len(self.pool))):
+            raise ValueError("Expected pool to contain keys 0...(N-1)")
+
+    def __call__(
+        self, mapped_triples: MappedTriples, num_tokens: int, num_entities: int, num_relations: int
+    ) -> Tuple[int, torch.LongTensor]:  # noqa: D102
+        if num_entities != len(self.pool):
+            raise ValueError(f"Invalid number of entities ({num_entities}); expected {len(self.pool)}")
+        return self.vocabulary_size, _random_sample_no_replacement(pool=self.pool, num_tokens=num_tokens)
```

Comment:
I think we don't need random sampling here and should rather select the first K anchors available in downloaded pools - even though we discard anchor distances, the anchors are still ordered `nearest to farthest` . 
Besides, that's what we also do in `CSGraphAnchorSearcher` and `ScipySparseAnchorSearcher`

Reply:
Okay, that's even simpler :sweat_smile:

Reply:
24faaf3